### PR TITLE
Fix for cards not showing up inside card browser if deck contains wildcard characters in its name.

### LINKED ISSUE #6288: Card Browser: Can't browse certain decks containing brackets and stars
###### Regression Test

Please use `*Yr1::Med2::CAS4::F4: Renal::BRS (zanki)::HY` as input. Works in Anki Desktop

###### Reproduction Steps

1. Name a deck `*Hello (zanki)`
2. Add a card
3. Card Browser

Note: `Hello (zanki)` and `*Hello` both work fine
###### Expected Result

4. Card is shown

###### Actual Result

4. Card is not shown

###### Debug info
Latest alpha

###### Research
- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by anadi198**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
Fixes #6288

## Approach
Previously, if there was a `*` character in the deck name, then exact matching of deck name was skipped over and regex search was applied. So, to fix that, an exact name search is done first, if it doesn't match any deck then we proceed to applying regex searching.

## How Has This Been Tested?

I tested it both with and without the modification on a physical API 29 device. I was able to reproduce the bug and with the modification the card is showing up now.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)

### CODE REVIEW

**Comment by anadi198** [comment]
I don't know how Project.xml got changed(probably due to updating Android Studio to build this project). I will add a commit reverting those changes. Apologies.

File: AnkiDroid/src/main/java/com/ichi2/libanki/Finder.java [hunk 1]
```diff
@@ -659,16 +659,20 @@ public String _findDeck(String val) {
...
-                    for (long id : dids(d.getLong("id"))) {
-                        if (!ids.contains(id)) {
-                            ids.add(id);
+            ids = dids(mCol.getDecks().id(val, false));
+            if (ids == null) {
+                ids = new ArrayList<>();
+                val = val.replace("*", ".*");
+                val = val.replace("+", "\\+");
+
+                for (JSONObject d : mCol.getDecks().all()) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Finder.java:668]: This was changed, but I don't think there was a reason for it

```suggestion
                for (Deck d : mCol.getDecks().all()) {
```

**Review by Arthur-Milchior — APPROVED** [review]
Are you saying that if a card contains "*" then it can't be found ? I would have thought that "*" would also match "*" even if it matches other things too.

The idea makes a lot of sens. Do an exact match if possible and not exact otherwise. It probably also makes more sens for people not used to globbing

**Review by david-allison-1 — APPROVED** [review]

**Comment by david-allison-1** [comment]
It'd be nice to have a regression test to ensure that there's no regressions for this issue. 

@anadi198 , would you like to take this on in a separate PR to get some more DigitalOcean swag points? 😉

**Comment by anadi198** [comment]
> It'd be nice to have a regression test to ensure that there's no regressions for this issue.
> 
> @anadi198 , would you like to take this on in a separate PR to get some more DigitalOcean swag points? 😉

For sure, but first I will need to study about regression tests(I have almost zero experience with writing tests). I'll probably make some mistakes or need suggestions. If someone else is more suitable for this then I'm fine with them doing it as well!

**Comment by anadi198** [comment]
@david-allison-1 One of the builds failed, is there something wrong with my PR?

**Comment by david-allison-1** [comment]
@anadi198 Not your fault - will look into it

**Comment by mikehardy** [comment]
@anadi198 we go to somewhat great lengths in CI to pre-fetch all network-located objects before the main build where failure is related to the PR instead of a network failure, but it is a constant battle as new network-located dependencies creep in all the time.

Looks like we have one here that caused the build failure, and it does not appear to have anything at all to do with your PR:

```
> Configure project :AnkiDroid

Checking the license for package Android SDK Build-Tools 29.0.2 in /home/travis/android-sdk/licenses

License for package Android SDK Build-Tools 29.0.2 accepted.

Preparing "Install Android SDK Build-Tools 29.0.2 (revision: 29.0.2)".

"Install Android SDK Build-Tools 29.0.2 (revision: 29.0.2)" ready.

Installing Android SDK Build-Tools 29.0.2 in /home/travis/android-sdk/build-tools/29.0.2

"Install Android SDK Build-Tools 29.0.2 (revision: 29.0.2)" complete.

"Install Android SDK Build-Tools 29.0.2 (revision: 29.0.2)" finished.

Checking the license for package Android SDK Platform 29 in /home/travis/android-sdk/licenses

License for package Android SDK Platform 29 accepted.

Preparing "Install Android SDK Platform 29 (revision: 5)".

"Install Android SDK Platform 29 (revision: 5)" ready.

Installing Android SDK Platform 29 in /home/travis/android-sdk/platforms/android-29

"Install Android SDK Platform 29 (revision: 5)" complete.

"Install Android SDK Platform 29 (revision: 5)" finished.

FAILURE: Build failed with an exception.

* What went wrong:

A problem occurred configuring project ':AnkiDroid'.

> Could not create task ':AnkiDroid:sdkReleaseDependencyData'.

   > Could not create task of type 'SdkDependencyDataGeneratorTask'.

      > com/google/crypto/tink/hybrid/HybridConfig

* Try:

Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 30s

The command "./gradlew :AnkiDroid:compileReleaseJavaWithJavac compileLint" failed. Retrying, 2 of 3.

```

I'll log a separate issue for it

In the meantime I cleared status on that unrelated failure and odds are it will complete just fine resulting in a fully green status here

**Comment by anadi198** [comment]
@mikehardy It failed again..

**Comment by mikehardy** [comment]
really sorry about that, still not related to this PR, related to something else added/changed recently

```
FAILURE: Build failed with an exception.

* What went wrong:

A problem occurred configuring project ':AnkiDroid'.

> Could not create task ':AnkiDroid:sdkReleaseDependencyData'.

   > Could not create task of type 'SdkDependencyDataGeneratorTask'.

      > com/google/crypto/tink/hybrid/HybridConfig

* Try:

Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 26s

```

**Comment by mikehardy** [comment]
merged! Thanks @anadi198 !
